src/Centreon/Infrastructure/Monitoring/ResourceRepositoryRDB.php

```diff
@@ -227,19 +267,61 @@ function ($acc, $concordanceKey) {
         return false;
     }
 
+    /**
+     * Check if the filters are compatible to extract services
+     *
+     * @return bool
+     */
+    private function hasServiceFilter(ResourceFilter $filter): bool
+    {
+        if (($filter->getTypes() && !$filter->hasType(ResourceFilter::TYPE_SERVICE)) ||
+            ($filter->getStatuses() && !ResourceFilter::map(
+                $filter->getStatuses(),
+                ResourceFilter::MAP_STATUS_SERVICE
+            ))
+        ) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Check if the filters are compatible to extract hosts
+     *
+     * @return bool
+     */
+    private function hasHostFilter(ResourceFilter $filter): bool
+    {
+        if ($this->hasServiceSearch() ||
+            ($filter->getTypes() && !$filter->hasType(ResourceFilter::TYPE_HOST)) ||
+            ($filter->getStatuses() && !ResourceFilter::map(
+                $filter->getStatuses(),
+                ResourceFilter::MAP_STATUS_HOST
+            )) ||
+            $filter->getServicegroupIds()
+        ) {
+            return false;
+        }
+
+        return true;
+    }
+
     /**
      * Prepare SQL query for services
      *
      * @param StatementCollector $collector
-     * @param array $filterState
+     * @param ResourceFilter $filter
      * @return string
      */
-    protected function prepareQueryForServiceResources(StatementCollector $collector, ?array $filterState): string
+    protected function prepareQueryForServiceResources(StatementCollector $collector, ResourceFilter $filter): string
     {
         $sql = "SELECT
             s.service_id AS `id`,
             'service' AS `type`,
+            's' AS `short_type`,
```

Comment:
for future, I think it's better to use virtual properties in serializer to avoid unneccessary job on database side